Lib/sqlite3/test/userfunctions.py

```diff
@@ -351,6 +298,24 @@ def test_empty_blob(self):
         cur = self.con.execute("select isblob(x'')")
         self.assertTrue(cur.fetchone()[0])
 
+    def test_func_params(self):
+        self.con.create_function("boomerang", 1, lambda x: x)
+        dataset = (
+            (42, int),
+            (1<<42, int),  # long long
+            (3.14, float),
+            ("text", str),
```

Comment:
While we are here, it would be nice to add tests for a string containing non-ASCII BMP characters (in the range U-0080 - U-FFFF) and for a string containing non-BMP characters (in the range U-FFFF - U-10FFFF).

Strings containing lone surrogate codes (high surrogates U-D800 - U-DBFF and low surrogates U-DC00 - U-DFFF) should be tested separately. In the current implementation they produce errors.

Reply:
Covered in #27654.

Reply:
#27654 added tests for result containing surrogates, but we need also tests for parameters containing surrogates, and for normal strings containing non-ASCII characters.